internal/config/config_file.go

```diff
@@ -8,23 +8,76 @@ import (
 	"os"
 	"path"
 	"path/filepath"
+	"runtime"
 	"syscall"
 
 	"github.com/mitchellh/go-homedir"
 	"gopkg.in/yaml.v3"
 )
 
 const (
-	GH_CONFIG_DIR = "GH_CONFIG_DIR"
+	GH_CONFIG_DIR   = "GH_CONFIG_DIR"
+	XDG_CONFIG_HOME = "XDG_CONFIG_HOME"
+	APP_DATA        = "AppData"
 )
 
+// Config path precedence
+// 1. GH_CONFIG_DIR
+// 2. XDG_CONFIG_HOME
+// 3. AppData (windows only)
+// 4. HOME
 func ConfigDir() string {
-	if v := os.Getenv(GH_CONFIG_DIR); v != "" {
-		return v
+	var path string
+	if a := os.Getenv(GH_CONFIG_DIR); a != "" {
+		path = a
+	} else if b := os.Getenv(XDG_CONFIG_HOME); b != "" {
+		path = filepath.Join(b, "gh")
+	} else if c := os.Getenv(APP_DATA); runtime.GOOS == "windows" && c != "" {
+		path = filepath.Join(c, "GitHub CLI")
+	} else {
+		d, _ := os.UserHomeDir()
+		path = filepath.Join(d, ".config", "gh")
+	}
+
+	// If the path does not exist try migrating config from default paths
+	if _, err := os.Stat(path); err != nil {
+		autoMigrateConfigDir(path)
+	}
+
+	return path
+}
+
+// Check default paths (os.UserHomeDir, and homedir.Dir) for existing configs
+// If configs exist then move them to newPath
+// TODO: Remove support for homedir.Dir location in v2
+func autoMigrateConfigDir(newPath string) {
+	path, err := os.UserHomeDir()
+	if oldPath := filepath.Join(path, ".config", "gh"); err == nil && dirExists(oldPath) {
+		migrateConfigDir(oldPath, newPath)
+		return
+	}
+
+	path, err = homedir.Dir()
+	if oldPath := filepath.Join(path, ".config", "gh"); err == nil && dirExists(oldPath) {
+		migrateConfigDir(oldPath, newPath)
 	}
+}
 
-	homeDir, _ := homeDirAutoMigrate()
-	return homeDir
+func dirExists(path string) bool {
+	f, err := os.Stat(path)
+	if err == nil && f.IsDir() {
+		return true
+	}
+	return false
+}
+
+var migrateConfigDir = func(oldPath, newPath string) {
+	if oldPath == newPath {
+		return
+	}
+
+	_ = os.MkdirAll(filepath.Dir(newPath), 0755)
+	_ = os.Rename(oldPath, newPath)
```

Comment:
Should we also remove the orphaned `~/.config` directory after the migration of `~/.config/gh` to the new location?

We could simply try to `os.Remove` the parent. If it errors out, we ignore it.

Reply:
I don't think this is a good idea since it could remove files users want to keep. I actually have `git`, `kitty`, and `nvim` configuration files in `~/.config`.

Reply:
`os.Remove`, like the `rm` command, is only able to remove empty directories. It would error out otherwise.